### ISSUE TITLE
French Date issue. Probably Regexp fail?

### ISSUE THREAD

**Issue by ChauskinRodion** [clemens/delocalize#77]
Hey Guys. 
I have a trouble with parsing :fr date "Févr. 12, 2015" to :en format.

Few examples from my console below:
```console
$: date
=> "Févr. 12, 2015"
$: Delocalize::LocalizedDateTimeParser.parse(date, Date)
=> Sun, 05 Apr 2015
$: I18n.l(DateTime.parse('Feb 12 2015'))
=> "12 février 2015 00h 00min 00s"
$: I18n.l(DateTime.parse('Feb 12 2015'), format: '%b %d, %Y')
=> "Févr. 12, 2015"
$: date =  I18n.l(DateTime.parse('Mar 12 2015'), format: '%b %d, %Y')
=> "Mars 12, 2015"
$: Delocalize::LocalizedDateTimeParser.parse(date, Date)
=> Sun, 05 Apr 2015
```
So, as you can see, something strange happens with my code :(. 

actually, i've implemented quickfix at my project to push it live, but i want to get more clear explanation  - where i'm wrong? 

here is my quickfix::
lib/extensions/localized_date_time_parser.rb
```ruby
class << Delocalize::LocalizedDateTimeParser

  private
  def translate_month_and_day_names(datetime)

    translated = [:month_names, :abbr_month_names, :day_names, :abbr_day_names].map do |key|
      I18n.t(key, :scope => :date)
    end.flatten.compact

    original = (Date::MONTHNAMES + Date::ABBR_MONTHNAMES + Date::DAYNAMES + Date::ABBR_DAYNAMES).compact
    #translated.each_with_index { |name, i| datetime.gsub!(/\b#{name}\b/, original[i]) }
    translated.each_with_index { |name, i| datetime.gsub!(/\b#{name}/, original[i]) }
  end

end
```

**Comment by clemens** [comment]
Could you post the YAML file where you've specified your date formats?

**Comment by ChauskinRodion** [comment]
@clemens , thank you for quick answer. apreciate it.
Yes, i have a custom yaml locale.. (here is some different with original rails-i18n date format)

fr:
  date:
    abbr_day_names:
      - dim
      - lun
      - mar
      - mer
      - jeu
      - ven
      - sam
    abbr_month_names:
      - ~
      - Janv.
      - Févr.
      - Mars
      - Avril
      - Mai
      - Juin
      - Juil.
      - Août
      - Sept.
      - Oct.
      - Nov.
      - Déc.
    day_names:
      - dimanche
      - lundi
      - mardi
      - mercredi
      - jeudi
      - vendredi
      - samedi
    formats:
      default: "%b %d, %Y"
      short: "%e %b"
      long: "%e %B %Y"
    month_names:
      - ~
      - janvier
      - février
      - mars
      - avril
      - mai
      - juin
      - juillet
      - août
      - septembre
      - octobre
      - novembre
      - décembre
    order:
      - :day
      - :month
      - :year

**Comment by clemens** [comment]
As it says in the readme (see https://github.com/clemens/delocalize#what-does-it-do-and-how-do-i-use-it), you need to specify an `input.formats` array so delocalize can actually figure out which formats to accept. If it can't find fitting formats, it just falls back to the current date and time. Yes, this is a little odd – but I needed this so I could parse dates without years so I thought, I might as well use all values of the current date and time as default.

In short: Make sure the keys `date.input.formats` and `time.input.formats` are defined (after you've added them to the YAML file, make sure `I18n.t('date.input.formats')` and `I18n.t('time.input.formats')` return proper values). Then everything should work.

**Comment by ChauskinRodion** [comment]
@clemens , great, thank you, sir! 
will try & share results with you.

**Comment by clemens** [comment]
Any news? Can I close this?

**Comment by clemens** [comment]
Any updates? Otherwise I'll close this.

**Comment by clemens** [comment]
Closing this for inactivity. Please comment if you want it reopened.
